apps/web/src/utils/config.ts

```diff
@@ -57,5 +57,28 @@ function has(predicate: (key: string) => boolean) {
   return false;
 }
 
-const Config = { set, get, clear, all, has, remove };
+function logout() {
+  const toKeep = [
+    "editorConfig",
+    "backupStorageLocation",
+    "serverUrls",
+    "corsProxy",
+    "theme:light",
+    "theme:dark",
+    "colorScheme",
+    "followSystemTheme"
+  ].reduce((acc, key) => {
+    const val = get(key);
+    if (val !== undefined) acc[key] = val;
+    return acc;
+  }, {} as Record<string, any>);
+
+  clear();
+
+  Object.entries(toKeep).forEach(([key, val]) => {
+    set(key, val);
+  });
```

Comment:
Can we simplify this? I don't think going full on functional here is improving things. Just use simple loops.